### PR TITLE
Add possibility for interceptors in the GrpcPods client

### PR DESCRIPTION
**Pull request by SvenW**
This will make it possible to for example add a client interceptor that handles tracing propagation between pods etc

### CODE REVIEW

File: protocol-grpc/src/main/scala/com/devsisters/shardcake/GrpcConfig.scala [hunk 1]
```diff
@@ -11,9 +12,14 @@ import java.util.concurrent.Executor
  * @param executor a custom executor to pass to grpc-java when creating gRPC clients and servers
  * @param shutdownTimeout the timeout to wait for the gRPC server to shutdown before forcefully shutting it down
  */
-case class GrpcConfig(maxInboundMessageSize: Int, executor: Option[Executor], shutdownTimeout: Duration)
+case class GrpcConfig(
+  maxInboundMessageSize: Int,
+  executor: Option[Executor],
+  shutdownTimeout: Duration,
+  interceptors: Seq[ZClientInterceptor]
```
Comment (ghostdogpr) [inline protocol-grpc/src/main/scala/com/devsisters/shardcake/GrpcConfig.scala:20]: Can you add the scaladoc for this one?

Comment (SvenW) [reply protocol-grpc/src/main/scala/com/devsisters/shardcake/GrpcConfig.scala:20]: > Can you add the scaladoc for this one?

Absolutely

Comment (ghostdogpr) [reply protocol-grpc/src/main/scala/com/devsisters/shardcake/GrpcConfig.scala:20]: Ah I meant the `@param` like the other parameters above

Comment (SvenW) [reply protocol-grpc/src/main/scala/com/devsisters/shardcake/GrpcConfig.scala:20]: Haha, of course. How could I have missed that 🤦

**Pull request merged**